Update Verifying-ERC20.md

amystrayer opened the pull request:
In line 14, is this a direct quote? I added an "and" as a correction, but if this is a direct quote, then leave it as you wrote it.
In line 87, "since it must *quantify* over the set of all accounts", Is over normally used after quantify in these types of documents? I feel the 'over' is unnecessary, but if it's normally used, then keep it. There's another instance in line 193.
In line 216, is "The problem is due to a bug in `mint`" an example of a type of error verisol would report? I'm trying to figure out the right punctuation.
In line 218, I changed 'between' to 'among'. Between is for 2 things, and among is for 3 or more things. I'm not sure which one is correct for you, so you can choose depending on how many variables you're talking about.
In line 246, is "instruments" being used as a verb here "The tool instruments the contract to find"?

Not sure why, but in some lines or words, it crossed out the entire thing and rewrote it, even if I just changed one thing.

amystrayer committed:
Update Verifying-ERC20.md

In line 14, is this a direct quote? I added an "and" as a correction, but if this is a direct quote, then leave it as you wrote it.
In line 87, "since it must *quantify* over the set of all accounts", Is over normally used after quantify in these types of documents? I feel the 'over' is unnecessary, but if it's normally used, then keep it. There's another instance in line 193.
In line 216, is "The problem is due to a bug in `mint`" an example of a type of error verisol would report? I'm trying to figure out the right punctuation.
In line 218, I changed 'between' to 'among'. Between is for 2 things, and among is for 3 or more things. I'm not sure which one is correct for you, so you can choose depending on how many variables you're talking about.
In line 246, is "instruments" being used as a verb here "The tool instruments the contract to find"?

Not sure why, but in some lines or words, it crossed out the entire thing and rewrote it, even if I just changed one thing.

garbervetsky commented:
Hi Amy! Thanks for all the corrections!  

> In line 14, is this a direct quote? I added an "and" as a correction, but if this is a direct quote, then leave it as you wrote it.

Yes. It was a a quote from here (https://forum.openzeppelin.com/t/online-erc20-contract-verifier/1575). I believe I should fix the link. 

> In line 87, "since it must _quantify_ over the set of all accounts", Is over normally used after quantify in these types of documents? I feel the 'over' is unnecessary, but if it's normally used, then keep it. There's another instance in line 193.

Yes, here "quantify"  is a logical operation over a set of elements. It usually reads quantify over.     

> In line 216, is "The problem is due to a bug in `mint`" an example of a type of error verisol would report? I'm trying to figure out the right punctuation.

"There is a bug (and error) in the function 'mint`. This is detected by a sequence of transactions (including the invocation of the function `mint`) that violates the given invariant.  
Notice that there is a minor issue in the current version of Verisol: Instead of reporting the line corresponding to the contract invariant tt wrongly reports the bug in the last line of the constructor (This [issue](https://github.com/microsoft/verisol/issues/166) has already been reported.)."

Is that better? :-) 

> In line 218, I changed 'between' to 'among'. Between is for 2 things, and among is for 3 or more things. I'm not sure which one is correct for you, so you can choose depending on how many variables you're talking about.

among is better! Actually it can 2 or more things. 

> In line 246, is "instruments" being used as a verb here "The tool instruments the contract to find"?
> 

Instrument is used as a  verb  :-). This is OK :-)

Thanks!

amystrayer commented:
Ok, everything looks fine then, except for one correction-
In the line 216 question, yes, what you wrote is fine with a few small
changes. You can replace this sentence:
Instead of reporting the line corresponding to the contract invariant, it
wrongly reports the bug in the last line of the constructor

Let me know for anything else!
Best,
Amy

On Mon, Dec 2, 2019 at 12:53 PM garbervetsky <notifications@github.com>
wrote:

> Hi Amy! Thanks for all the corrections!
>
> In line 14, is this a direct quote? I added an "and" as a correction, but
> if this is a direct quote, then leave it as you wrote it.
>
> Yes. It was a a quote from here (
> https://forum.openzeppelin.com/t/online-erc20-contract-verifier/1575). I
> believe I should fix the link.
>
> In line 87, "since it must *quantify* over the set of all accounts", Is
> over normally used after quantify in these types of documents? I feel the
> 'over' is unnecessary, but if it's normally used, then keep it. There's
> another instance in line 193.
>
> Yes, here "quantify" is a logical operation over a set of elements. It
> usually reads quantify over.
>
> In line 216, is "The problem is due to a bug in mint" an example of a
> type of error verisol would report? I'm trying to figure out the right
> punctuation.
>
> "There is a bug (and error) in the function 'mint. This is detected by a
> sequence of transactions (including the invocation of the function mint`)
> that violates the given invariant.
> Notice that there is a minor issue in the current version of Verisol:
> Instead of reporting the line corresponding to the contract invariant tt
> wrongly reports the bug in the last line of the constructor (This issue
> <https://github.com/microsoft/verisol/issues/166> has already been
> reported.)."
>
> Is that better? :-)
>
> In line 218, I changed 'between' to 'among'. Between is for 2 things, and
> among is for 3 or more things. I'm not sure which one is correct for you,
> so you can choose depending on how many variables you're talking about.
>
> among is better! Actually it can 2 or more things.
>
> In line 246, is "instruments" being used as a verb here "The tool
> instruments the contract to find"?
>
> Instrument is used as a verb :-). This is OK :-)
>
> Thanks!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/garbervetsky/ERC-Verisol-Demo/pull/3?email_source=notifications&email_token=AMOMYMK7PJIBRMWKH2UCK2DQWUVN3A5CNFSM4JTWQ5GKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFT6BNY#issuecomment-560455863>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMOMYMLN5XR2FTHRYR7EPH3QWUVN3ANCNFSM4JTWQ5GA>
> .
>

garbervetsky commented:
Done!

Thanks!

On Mon, Dec 2, 2019 at 9:04 PM amystrayer <notifications@github.com> wrote:

> Ok, everything looks fine then, except for one correction-
> In the line 216 question, yes, what you wrote is fine with a few small
> changes. You can replace this sentence:
> Instead of reporting the line corresponding to the contract invariant, it
> wrongly reports the bug in the last line of the constructor
>
> Let me know for anything else!
> Best,
> Amy
>
> On Mon, Dec 2, 2019 at 12:53 PM garbervetsky <notifications@github.com>
> wrote:
>
> > Hi Amy! Thanks for all the corrections!
> >
> > In line 14, is this a direct quote? I added an "and" as a correction, but
> > if this is a direct quote, then leave it as you wrote it.
> >
> > Yes. It was a a quote from here (
> > https://forum.openzeppelin.com/t/online-erc20-contract-verifier/1575). I
> > believe I should fix the link.
> >
> > In line 87, "since it must *quantify* over the set of all accounts", Is
> > over normally used after quantify in these types of documents? I feel the
> > 'over' is unnecessary, but if it's normally used, then keep it. There's
> > another instance in line 193.
> >
> > Yes, here "quantify" is a logical operation over a set of elements. It
> > usually reads quantify over.
> >
> > In line 216, is "The problem is due to a bug in mint" an example of a
> > type of error verisol would report? I'm trying to figure out the right
> > punctuation.
> >
> > "There is a bug (and error) in the function 'mint. This is detected by a
> > sequence of transactions (including the invocation of the function mint`)
> > that violates the given invariant.
> > Notice that there is a minor issue in the current version of Verisol:
> > Instead of reporting the line corresponding to the contract invariant tt
> > wrongly reports the bug in the last line of the constructor (This issue
> > <https://github.com/microsoft/verisol/issues/166> has already been
> > reported.)."
> >
> > Is that better? :-)
> >
> > In line 218, I changed 'between' to 'among'. Between is for 2 things, and
> > among is for 3 or more things. I'm not sure which one is correct for you,
> > so you can choose depending on how many variables you're talking about.
> >
> > among is better! Actually it can 2 or more things.
> >
> > In line 246, is "instruments" being used as a verb here "The tool
> > instruments the contract to find"?
> >
> > Instrument is used as a verb :-). This is OK :-)
> >
> > Thanks!
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/garbervetsky/ERC-Verisol-Demo/pull/3?email_source=notifications&email_token=AMOMYMK7PJIBRMWKH2UCK2DQWUVN3A5CNFSM4JTWQ5GKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFT6BNY#issuecomment-560455863
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AMOMYMLN5XR2FTHRYR7EPH3QWUVN3ANCNFSM4JTWQ5GA
> >
> > .
> >
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/garbervetsky/ERC-Verisol-Demo/pull/3?email_source=notifications&email_token=ACPRYJBCCHDJFXBHJTZSQA3QWWPBDA5CNFSM4JTWQ5GKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFXTPBY#issuecomment-560936839>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACPRYJDOT3PCXLJ7ELAVOF3QWWPBDANCNFSM4JTWQ5GA>
> .
>
